.github/workflows/build-and-deploy.yml

```diff
@@ -164,27 +164,23 @@ jobs:
 
   deploy-before-production:
     name: Parallel deployment before production
-    environment:
-      name: ${{ matrix.environment }}
-      url: ${{ steps.deploy_app_before_production.outputs.deploy-url }}
-    if: ${{ success() && github.ref == 'refs/heads/main' }}
+    if: ${{ github.ref == 'refs/heads/main' }}
     needs: [test, next_academic_year_test]
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         environment: [qa, staging]
     steps:
-    - name: Checkout
-      uses: actions/checkout@v4
-
-    - name: Deploy app to ${{ matrix.environment }}
-      id: deploy_app_before_production
-      uses: ./.github/actions/deploy/
+    - name: Deploy before production
+      uses: ./.github/actions/deploy-before-production
       with:
-        azure-credentials: ${{ secrets[format('AZURE_CREDENTIALS_{0}', matrix.environment)] }}
         environment: ${{ matrix.environment }}
-        sha: ${{ github.sha }}
+        # NOTE: Uncomment the below and make it required changes
+        #       Can use `@twd-register-devs` and such to inform,
+        # skip-environments: [qa]
+        # skip-environments-message: 'PUT SOMETHING MEANINGFUL'
```

Comment:
Maybe move it in to secrets altogether

Reply:
yes, it would be better in a secret so you don't have to merge a PR.
I would check how TRS do this. If their preprod is not from the main branch they skip it automatically. see https://github.com/DFE-Digital/teaching-record-system/blob/main/.github/workflows/main.yml#L29 although this method does have an impact on later jobs that refer to earlier jobs i.e. https://github.com/DFE-Digital/teaching-record-system/blob/main/.github/workflows/main.yml#L94